project/Sheng08/README.md

```diff
@@ -0,0 +1,253 @@
+# OrientMesh Toolkit: Extracting and Visualizing Oriented Bounding Boxes from 3D Triangular Meshes
+
+## Basic Information
+
+> **Project Overview:** Extracting and Visualizing Oriented Bounding Boxes (OBB) from Triangular Meshes using C++ and Python
+
+Triangular meshes[^1] stand as a cornerstone in computer graphics, essential in the portrayal and animation of 3D objects. A critical aspect of processing these meshes involves extracting **Oriented Bounding Boxes (OBB)**[^2]. _++OrientMesh Toolkit++_ presents an optimal solution to this task, fusing efficient algorithms with an interactive visual interface. This combination not only simplifies OBB extraction but also enhances user comprehension through vivid visual representations.
+
+- 📂 **GitHub Repository**: [https://github.com/Sheng08/OrientMesh-Toolkit](https://github.com/Sheng08/OrientMesh-Toolkit)
+- 💻 **Implemented In**:
+  - **Core Library**: C++
+  - **Bindings & Interface(GUI)**: Python
+
+## Problem to Solve
+
+Triangular meshes, due to their intricate detailing, can sometimes be complex and computationally heavy for operations. Having a precise oriented bounding box aids in simplifying many computational geometry problems, like collision detection, mesh simplification, and many more. However, generating an optimal OBB is a non-trivial task. OrientMesh aims to offer a reliable, optimized, and easily-integrable solution for this challenge.
+
+<p align="center">
+    <img src="./doc/pic/OOB_mesh_sample1.png" alt="" height="250"style="margin-right: 15px;"/>
+    <img src="./doc/pic/OOB_mesh_sample2.png" alt="" height="250"/>
+</p>
+
+## Prospective Users
+
+_++OrientMesh Toolkit++_ targets game developers, computer science students, researchers, and professionals working with 3D models, especially in areas requiring accurate bounding boxes like game development, simulation, and augmented reality.
+
+## System Architecture
+
+_++OrientMesh Toolkit++_ utilizes C++ for its core computational tasks, ensuring performance optimization. Moreover, Python is employed for its bindings, making it seamlessly usable for a wider audience and also aiding in GUI development and design.
+
+**Components**:
+
+- **Triangular Mesh Data Structures (C++)**: This component serves as the foundation, hosting the mesh structures essential for subsequent operations.
```

Comment:
The mesh data structure is the foundation.  Do you have an idea how do you want to design it?

Reply:
Yes, I have a preliminary design in mind for it. Here's a rough sketch of the data structures I'm considering:

```cpp
struct Vertex {
    glm::vec3 coordinates;
    glm::vec3 normal;       // Optional
};

struct Edge {
    Vertex* start;
    Vertex* end;
    std::vector<Face*> faces;
};

struct Face {
    Vertex* vertices[3];
    Edge* edges[3];
    glm::vec3 normal;       // This will facilitate operations that require face normals.
};

class MeshData {
    std::vector<Vertex> vertices;
    std::vector<Edge> edges;
    std::vector<Face> faces;

    // Additional functions for manipulating and querying the mesh
};
```

Reply:
Please add it somewhere in your proposal or point it out in your PR.  I can't find it in the updated proposal.